### ISSUE TITLE
Training: multi-GPU run-directory collision and incorrect masked latent MSE

### ISSUE THREAD

**Issue by Kevinsvip888** [huawei-bayerlab/marigold-v2#6]
Hi,

Congratulations on the great work.  When I tried the project on my own, I ran into some issues:

## 1. Multi-GPU startup fails when ranks create the same run directory

**Location:** `marigoldv2/script/train/setup_run.py`, line 77; called from `marigoldv2/script/train/train.py`, line 50. Line numbers refer to the inspected local copy.

Every rank calls `setup_run()` before `Accelerator` is initialized and executes:

```python
os.makedirs(out_dir_run, exist_ok=False)
```

With the default run naming, all ranks use the same path. One creates it and another raises `FileExistsError`. The `is_rank0` guard only protects later configuration saving, not this directory creation.

**Trigger:** Starting a fresh run with the documented `accelerate launch --multi_gpu ...` workflow and the same output directory/configuration on each rank, without a datetime prefix.

**Test performed:** Two independent processes executed the repository's actual `setup_run()` and `clean_folder_name()` functions, synchronizing immediately before run-directory creation. Configuration loading, logging, GPU detection, and code snapshots were stubbed.

**Result:** In all three trials, exactly one process succeeded and the other raised `FileExistsError` at line 77. Either rank could lose the race.

## 2. Masked latent MSE mishandles five-dimensional latents

**Location:** `marigoldv2/loss/loss.py`, `MSELoss._build_latent_mask()` and `MSELoss.__call__()`, approximately lines 207-244.

Qwen predictions and targets have shape `[B, C, 1, H, W]`, but the generated mask has shape `[B, 1, H, W]`. The channel-expansion branch is skipped because the tensors have different dimensionality. The reduction is then:

```python
loss = (diff * latent_mask).sum() / latent_mask.sum().clamp(min=1.0)
```

For `B=1`, broadcasting succeeds, but the numerator sums over channels while the denominator only counts valid spatial positions. The loss and gradients are scaled by the number of latent channels relative to elementwise mean squared error. For `B=2, C=16`, broadcasting fails because the mask's batch dimension aligns with the prediction's channel dimension.

**Comment by Igzi** [comment]
Thanks for highlighting these issues and for taking the time to fix the multi-GPU training problem.

We reviewed your PR, approved it, and have now merged the changes into the main branch. Thanks again for the contribution and for helping improve the training code!
